src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -72,7 +72,8 @@ The .NET Foundation licenses this file to you under the MIT license.
     <ItemGroup Condition="'$(StaticICULinking)' == 'true'">
       <NativeLibrary Include="$(IntermediateOutputPath)/libs/System.Globalization.Native/build/libSystem.Globalization.Native.a"/>
       <DirectPInvoke Include="libSystem.Globalization.Native" />
-      <StaticICULibs Include="-Wl,-Bstatic -licuio -licutu -licui18n -licuuc -licudata -Wl,-Bdynamic" />
+      <StaticICULibs Include="-Wl,-Bstatic -licuio -licutu -licui18n -licuuc -licudata -lstdc++" />
+      <StaticICULibs Include="-Wl,-Bdynamic" Condition="'$(NativeLib)' != 'Static' and !@(LinkerArg->Contains('-static'))" />
```

Comment:
Should `-Wl,-Bstatic` on previous line get the same condition for symmetry?

Reply:
We can use `StaticICULinking=true` with dynamically linked binary as well. I tested with and without `<LinkerArg Include="-static" />` line in csproj from #79498 repro. Also tested `<OutputType>Library` && `<NativeLib>Static`.

Reply:
When we are linking with `-static`, `-Wl,-Bstatic` should be unnecessary as well. Do you agree? 

My point was that we omit both `-Wl,-Bdynamic` and `-Wl,-Bstatic` flips for `-static`.

Reply:
Absolutely, updated. I did a quick test and confirmed it is not necessary.

The original problem was that with `-Wl,-Bdynamic` (which later gets prepended with `-pie` etc.), setting `-static` in user project was still producing `dynamically linked` executable and it was requiring to manually set the correct interpreter on musl: `<LinkerArg Include="-Wl,--dynamic-linker,/lib/ld-musl-aarch64.so.1" />` (to override the wrong/default one `ld-musl-aarch64.so.1`). Removing `-Wl,-Bdynamic` in that case made it `statically linked`.

Reply:
Absolutely, updated. I did a quick test and confirmed it is not necessary.

The original problem was that with `-Wl,-Bdynamic` (which later gets prepended with `-pie` etc.), setting `-static` in user project was still producing `dynamically linked` executable and it was requiring manual setting of correct interpreter on musl: `<LinkerArg Include="-Wl,--dynamic-linker,/lib/ld-musl-aarch64.so.1" />` (to override the wrong/default one `ld-musl-aarch64.so.1`). Removing `-Wl,-Bdynamic` in that case made it `statically linked`.

Reply:
Absolutely, updated. I did a quick test and confirmed it is not necessary.

The original problem was that with `-Wl,-Bdynamic` (which later gets prepended with `-pie` etc.), setting `-static` in user project was still producing `dynamically linked` executable and it was requiring manual setting of correct interpreter on musl: `<LinkerArg Include="-Wl,--dynamic-linker,/lib/ld-musl-aarch64.so.1" />` (to override the wrong/default one `ld-linux-aarch64.so.1`). Removing `-Wl,-Bdynamic` in that case made it `statically linked`.